### PR TITLE
DetailsList: Add isCollapsible and deprecate isCollapsable

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

I noticed that we have "collapsible" in all other places in the code, but we have "isCollapsable" as a property of IColumn (spelled with an "a"), which is not consistent. This PR adds some coherence.

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
@JasonGore, curious.. do we have an example of how to warn deprecation for properties which are not part of a component, but rather an interface used by the component? For example, in this PR, the deprecated property is present on `IColumn` not `DetailsList`.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx [hunk 1]
```diff
@@ -1052,7 +1052,7 @@ export function buildColumns(
           fieldName: propName,
           minWidth: MIN_COLUMN_WIDTH,
           maxWidth: 300,
-          isCollapsable: !!columns.length,
+          isCollapsible: !!columns.length,
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx]: You should continue setting isCollapsable until v7--there's a very good chance something downstream is reading it.

**Comment by JasonGore** [comment]
> @JasonGore, curious.. do we have an example of how to warn deprecation for properties which are not part of a component, but rather an interface used by the component? For example, in this PR, the deprecated property is present on `IColumn` not `DetailsList`.

Unless it is an internal API used by the component and not exposed I think it should use the same deprecation standard as a component API. It has the same capability of breaking builds. A pretty good litmus test for this in general is whether or not the interface appears in the generated API file.

**Comment by natalieethell** [comment]
@JasonGore I think what Kevin was asking was how we can warn deprecation for a property that is not a direct property of the component props interface. I know that we can warn deprecation for a property of the component props interface in the constructor, but what would we do in the case for `isCollapsible`, which is a property of IColumn, not a property of IDetailsListProps?

**Comment by JasonGore** [comment]
Oh sorry. It's an interface that consumers must implement, so I don't think there's any practical way to warn on usage.

**Comment by natalieethell** [comment]
@JasonGore okay thanks!

@khmakoto regarding your comment, I more so just meant that the spelling of collapsible/collapsable was with an "i" in all other places of the code. But good point, we actually have `isCollapsible` on the CodeBlock props interface and `collapsible` on a component props interface in experiments. So I think we could go either way.

**Comment by KevinTCoughlin** [comment]
> 
> 
> @JasonGore okay thanks!
> 
> @khmakoto regarding your comment, I more so just meant that the spelling of collapsible/collapsable was with an 'i' in all other places of the code. But good point, we actually have `isCollapsible` on the CodeBlock props interface and `collapsible` on a component props interface in experiments. So I think we could go either way.

Personally, I'd stick with the `is*` prefix for this change given other occurrences across OUFR. It conveys the type is `boolean` for me, but that's just one data point 😄. Agreed re: the mismatch, we'll try to address that going forward.

**Pull request merged**